Inferences/Induction.cpp

```diff
@@ -630,6 +630,7 @@ void InductionClauseIterator::performStructInductionThree(Clause* premise, Liter
       while(vtit.hasNext()){
         smallers = new FormulaList(new AtomicFormula(Literal::create1(sty,true,TermList(vtit.next(),false))),smallers);
       }
```

Comment:
actually this might be dubious, let me get back to this one

Reply:
Adding in some assertions where you "think" they should be true (but with a comment on why) is reasonable defensive programming - if we hit the assertion and think it's okay then we relax or remove it later.

Reply:
Actually this assert has to be true or we will crash - in the ternary operator below, either `smallers` has some length more than one (and is therefore non-null), or we call `->head()` - which will crash if `smallers` is `nullptr`.